### PR TITLE
Add typescript definitions

### PR DESCRIPTION
**Pull request by gpbl**
- [X] add `react-day-picker.d.ts` [from DefinitelyTyped]( https://github.com/giladgray/DefinitelyTyped/blob/266e2977fdd25e3618695121917e65dc6364c96a/types/react-day-picker/index.d.ts)
- [X] add entry in package.json
- [x] upgrade to last version (don't forget https://github.com/gpbl/react-day-picker/pull/296)
- [x] add dtslint <u>tests (see what [moment](https://github.com/moment/moment) is doing)</u>

See https://github.com/gpbl/react-day-picker/issues/299 and https://github.com/DefinitelyTyped/DefinitelyTyped/pull/16010

### CODE REVIEW

File: react-day-picker.d.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
+// Type definitions for react-day-picker 5.2
+// Project: https://github.com/gpbl/react-day-picker
+// Definitions by: Giampaolo Bellavite <https://github.com/gpbl>, Jason Killian <https://github.com/jkillian>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
```
Comment (adidahiya) [inline react-day-picker.d.ts]: I don't think you need the above 4 lines, just the TypeScript compiler version

File: package.json [hunk 2]
```diff
@@ -62,6 +62,7 @@
   "peerDependencies": {
     "react": "~0.13.x || ~0.14.x || ^15.0.0"
   },
+  "types": "./react-day-picker.d.ts",
```
Comment (giladgray) [inline package.json]: i'd say this belongs up near `main` and `style` keys as it's a distributable file

File: react-day-picker.d.ts [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        pagedNavigation?: boolean;
+        renderDay?(date: Date, modifiers: Modifiers): React.ReactNode;
+        reverseMonths?: boolean;
+        selectedDays?: Modifier | Modifier[];
+        toMonth?: Date;
+        weekdayElement?: React.ReactElement<Partial<WeekdayElementProps>> |
+            React.ComponentClass<WeekdayElementProps> |
+            React.SFC<WeekdayElementProps>;
+        weekdaysLong?: [string, string, string, string, string, string, string];
+        weekdaysShort?: [string, string, string, string, string, string, string];
```
Comment (giladgray) [inline react-day-picker.d.ts]: real game changing feature would be to include `/** docs */` on each of these props. i know you have great docs for every prop already... but that could easily be added in a future release.

File: docs/API.md [hunk 4]
```diff
@@ -1,47 +1,4 @@
 # Component API
 
 * [Component props](APIProps.md)
-* [Component methods](APIMethods.md)
-
-## Props quick reference
```
Comment (adidahiya) [inline docs/API.md]: I think you might want to add a link to types/index.d.ts here? People might not know to look for that immediately

Comment (gpbl) [inline docs/API.md]: Sure 👍🏽 However, aren't the definitions inferred by the tools using them?

PS. thanks @adidahiya @giladgray for the help! I never used TypeScript and I'm happy to learn something new!

File: package.json [hunk 5]
```diff
@@ -4,18 +4,21 @@
...
   "main": "DayPicker.js",
   "style": "lib/style.css",
+  "types": "./types/index.d.ts",
   "files": [
     "DayPicker.js",
     "lib",
     "moment.js",
-    "utils.js"
+    "utils.js",
+    "./types/index.d.ts"
```
Comment (giladgray) [inline package.json]: for consistency i'd omit the `./` but whatever 🚢

File: docs/API.md (hunk 4 above)
Comment (giladgray) [inline docs/API.md]: not exactly: the tools rely on package.json `"types"` key to tell them where to find an entry `.d.ts` file, and the `node_modules/@types` scope directory to list all the available external modules.

**Pull request merged**